h2o-algos/src/main/java/hex/svm/LLMatrix.java

```diff
@@ -0,0 +1,87 @@
+package hex.svm;
+
+import water.Iced;
+import water.MemoryManager;
+
+class LLMatrix extends Iced<LLMatrix> {
+
+  private final int _dim;
+  private final double[][] _data;
+
+  LLMatrix(int dim) {
+    _dim = dim;
+    _data = new double[dim][];
+    for (int i = 0; i < dim; i++) {
+      _data[i] = new double[dim - i];
+    }
+  }
+
```

Comment:
If I assume for the double array data, the first index is row index, the second index is column index, then the data is a upper triangular matrix, right?